src/main/java/amazin/model/Role.java

```diff
@@ -0,0 +1,36 @@
+package amazin.model;
+
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.Set;
+
+@Entity
+@Indexed
+@Table(name="role")
+public class Role {
+
+    @Id
```

Comment:
this is fine as long as it is unique which in our case it is

Reply:
we don't need an int id since it will never be used, user_name is the primary key here

Reply:
> this is fine as long as it is unique which in our case it is

> we don't need an int id since it will never be used, user_name is the primary key here

I see what you mean. A part of me thinks then table role does not add much if it just keeps a track record of who are the ADMINS and who are the USERS. In a perfect world, I believe this table could be just replaced with a column in the user table but for the time being that should be fine

Reply:
this wouldn't work because we are assuming a user can have multiple roles and as such we will have data duplication in the table

Reply:
> > this is fine as long as it is unique which in our case it is
> 
> > we don't need an int id since it will never be used, user_name is the primary key here
> 
> I see what you mean. A part of me thinks then table role does not add much if it just keeps a track record of who are the ADMINS and who are the USERS. In a perfect world, I believe this table could be just replaced with a column in the user table but for the time being that should be fine

this wouldn't work because we are assuming a user can have multiple roles and as such we will have data duplication in the table

Reply:
Well I also mentioned that users shouldnt have multiple roles you cant be ADMIN and a USER and an ANNONYMOUS that might not make sense how can i be annonymous and an admin?

Reply:
> Well I also mentioned that users shouldnt have multiple roles you cant be ADMIN and a USER and an ANNONYMOUS that might not make sense how can i be annonymous and an admin?

Aaron did this because he was thinking we might add more roles in the future

Reply:
This way is more abstract and therefore preferred. Abstraction is preferable in places where it makes sense. An example for future roles: a seller may also be a buyer. The rules you mentioned are just rules to be applied to the mapping.